enginetest/enginetests.go

```diff
@@ -4401,6 +4401,25 @@ func TestPreparedInsert(t *testing.T, harness Harness) {
 				},
 			},
 		},
+		{
+			Name: "simple decimal type insert",
+			SetUpScript: []string{
+				"CREATE TABLE test(id int primary key auto_increment, decimal_test DECIMAL(9,2), decimal_test_2 DECIMAL(9,2), decimal_test_3 DECIMAL(9,2))",
+			},
+			Assertions: []queries.ScriptTestAssertion{
+				{
+					Query: "INSERT INTO test(decimal_test, decimal_test_2, decimal_test_3) VALUES (?, ?, ?)",
+					Bindings: map[string]sql.Expression{
+						"v1": expression.NewLiteral(10, sql.Int64),
+						"v2": expression.NewLiteral([]byte("10.5"), sql.MustCreateString(sqltypes.VarBinary, 4, sql.Collation_binary)),
```

Comment:
Does the `varbinary` type trigger the same issue as `X'10'`? Just curious if it's worth adding that hex value to the test.

Reply:
Do you mean to test with `X'10'` instead of `10.5` as binding value?

Something I noticed that I did not understand clearly was that we convert many types into `varbinary` in https://github.com/dolthub/vitess/blob/main/go/mysql/query.go#L842 , but we are not able to convert them back to its original type in this method, https://github.com/dolthub/go-mysql-server/blob/main/server/handler.go#L180 , so for example passing in `X'10'` causes panic because we try to handle it as the same way as any string like `"X'10'"`

Reply:
Interesting... sending a hex string as a bindvar still causes a panic? Could you please open a tracking issue for that one with a good repro and we can tackle it separately?

Reply:
made an issue here https://github.com/dolthub/dolt/issues/4989